### ISSUE TITLE
Add `DataValidator` option for warning log as single DataFrame

### ISSUE THREAD

**Issue by dc-almeida** [IAMconsortium/nomenclature#438]
Current implementation of warning log for `DataValidator` prints a DataFrame with flagged rows for each criterion.
Suggested by @danielhuppmann to join all warnings into a single DataFrame, ordered by warning level (descending).
Question: what is the best format for the criteria messages (which would not be joined too, I presume)?

**Comment by danielhuppmann** [comment]
Good point, and after sleeping over it, I think the best solution would be to keep the log-output as is (stating the criteria, showing the failing data) but have an optional output of all failing criteria as one joined pandas.DataFrame (optionally saved to xlsx). This way, the log output is readable but if someone wants to look at the validation-results in a structured way, they have one nice dataframe.
